openedx/core/djangoapps/crawlers/models.py

```diff
@@ -39,6 +39,11 @@ def is_crawler(cls, request):
         if (not req_user_agent) or (not crawler_agents):
             return False
 
+        # If crawler agent type mismatches with the req_user_agent type then
+        # cast crawler agent to the type of req_user_agent.
+        if not isinstance(crawler_agents[0], type(req_user_agent)):
+            crawler_agents = [crawler_agent.encode('iso-8859-1') for crawler_agent in crawler_agents]
```

Comment:
This line might be confusing for folks who have to do maintenance later, so maybe a more detailed comment would help. Maybe something like:

```python
# The crawler_agents list we pull from our model always has unicode objects, but the
# req_user_agent we get from HTTP headers ultimately comes to us via WSGI. That
# value is an ISO-8859-1 encoded byte string in Python 2.7 (and in the HTTP spec), but
# it will be a unicode str when we move to Python 3.x. This code should work under
# either version.
if isinstance(req_user_agent, six.binary_type):
    crawler_agents = [crawler_agent.encode('iso-8859-1') for crawler_agent in crawler_agents]
```